contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj

```diff
@@ -0,0 +1,120 @@
+  "Check that the file exists"
+  [input-file]
+  (.exists (io/file input-file)))
+
+(def cli-options
+  [["-m" "--model-path-prefix PREFIX" "Model path prefix"
+    :default "models/resnet50_ssd/resnet50_ssd_model"
+    :validate [#(check-valid-file (str % "-symbol.json"))
+               "Model path prefix is invalid"]]
+   ["-i" "--input-image IMAGE" "Input image"
+    :default "images/kitten.jpg"
+    :validate [check-valid-file "Input file not found"]]
+   ["-d" "--input-dir IMAGE_DIR" "Input directory"
+    :default "images/"
+    :validate [check-valid-dir "Input directory not found"]]
+   ["-h" "--help"]])
+
+(defn print-predictions
+  "Print image detector predictions for the given input file"
+  [predictions width height]
+  (println (apply str (repeat 80 "=")))
+  (doseq [[label prob-and-bounds] predictions]
+    (println (format
+              "Class: %s Prob=%.5f Coords=(%.3f, %.3f, %.3f, %.3f)"
+              label
+              (aget prob-and-bounds 0)
+              (* (aget prob-and-bounds 1) width)
+              (* (aget prob-and-bounds 2) height)
+              (* (aget prob-and-bounds 3) width)
+              (* (aget prob-and-bounds 4) height))))
+  (println (apply str (repeat 80 "="))))
+
+(defn detect-single-image
+  "Detect objects in a single image and print top-5 predictions"
+  [detector input-image]
+  (let [image (infer/load-image-from-file input-image)
+        topk 5
+        [predictions] (infer/detect-objects detector image topk)]
+    predictions))
+
+(defn detect-images-in-dir
+  "Detect objects in all jpg images in the directory"
+  [detector input-dir]
+  (let [batch-size 20
+        image-file-batches (->> input-dir
+                                io/file
+                                file-seq
+                                (filter #(.isFile %))
+                                (filter #(re-matches #".*\.jpg$" (.getPath %)))
+                                (mapv #(.getPath %))
+                                (partition-all batch-size))]
+    (apply
+     concat
+     (for [image-files image-file-batches]
+       (let [image-batch (infer/load-image-paths image-files)
+             topk 5]
+         (infer/detect-objects-batch detector image-batch topk))))))
+
+(defn run-detector
+  "Runs an image detector based on options provided"
+  [options]
+  (let [{:keys [model-path-prefix input-image input-dir
+                device device-id]} options
+        width 512 height 512
+        descriptors [(mx-io/data-desc {:name "data"
+                                       :shape [1 3 height width]
+                                       :layout layout/NCHW
+                                       :dtype dtype/FLOAT32})]
+        factory (infer/model-factory model-path-prefix descriptors)
+        detector (infer/create-object-detector
+                  factory
+                  {:contexts [(context/default-context)]})]
+    (print-predictions (detect-single-image detector input-image) width height)
```

Comment:
Maybe some extra printing here to note on what is going on - ex "Object detection on single image" , "Object detection on images in a directory

Reply:
done